template/src/pageComponents/mint/steps/StartMintStep.tsx

```diff
@@ -12,30 +12,36 @@ type StartMintProps = {
 };
 
 export default function StartMintStep({ setMintStep }: StartMintProps) {
-  const { chain } = useNetwork();
-  const { address } = useAccount();
+  const { address, chain } = useAccount();
 
   const contract = useCustom1155Contract();
 
   const onCorrectNetwork = chain?.id === EXPECTED_CHAIN.id;
 
-  const { config } = usePrepareContractWrite({
+  const { data } = useSimulateContract({
     address: contract.status === 'ready' ? contract.address : undefined,
     abi: contract.abi,
     functionName: 'mint',
     args: address ? [address, BigInt(1), BigInt(1), address] : undefined,
-    enabled: onCorrectNetwork,
+    query: {
+      enabled: onCorrectNetwork,
+    },
   });
 
   // A future enhancement would be to use the `isLoading` and `isSuccess`
   // properties returned by `useContractWrite` to indicate transaction
   // status in the UI.
-  const { write: performMint } = useContractWrite({
-    ...config,
-    onSuccess() {
```

Comment:
I believe we should handle the `onSuccess` and `onError` state here as well

Reply:
`onSuccess` and `onError` callbacks have been removed in wagmi@2

Reply:
@eragon512 you can follow this pattern, I did [here](https://github.com/coinbase/build-onchain-apps/pull/338/files#diff-b1bd0f0d1a27dc6af7559a99bb29dcfdc796a1467a1b877fa6df93da685d6819R58)

Let me know if this makes sense!

Reply:
@Sneh1999 updated with a similar pattern - need some help with verifying the smart contract behaviour though